lib/api/hostendpoint.go

```diff
@@ -77,6 +77,9 @@ type HostEndpointSpec struct {
 	// profile is applied in the order that they appear in this list.  Profile rules are applied
 	// after the selector-based security policy.
 	Profiles []string `json:"profiles,omitempty" validate:"omitempty,dive,namespacedname"`
+
+	// Ports contains the endpoint's named ports, which may be referenced in security policy rules.
+	Ports []EndpointPort `json:"ports,omitempty" validate:"omitempty,dive"`
```

Comment:
That just tells the validator to examine each EndpointPort in the slice too.  It knows how to do that because we register a validator for EndpointPort too.

Reply:
For context: initially, I failed to add the dive and the validator ignored the Ports validation, so my new UTs failed.